ext/nkf/nkf.c

```diff
@@ -28,17 +28,30 @@
 #undef FALSE
 #define putchar(c)      rb_nkf_putchar(c)
 
+#ifndef RB_THREAD_LOCAL_SPECIFIER
+#  if __STDC_VERSION__ >= 201112
+#    define RB_THREAD_LOCAL_SPECIFIER _Thread_local
+#  elif defined(__GNUC__)
+     /* note that ICC (linux) and Clang are covered by __GNUC__ */
+#    define RB_THREAD_LOCAL_SPECIFIER __thread
+#  endif
+#endif
```

Comment:
This logic is borrowed from the `thread_pthread.h` file. On Ruby master it was removed in https://github.com/ruby/ruby/commit/f99af439805fe51b38aa1cf79f9c657f857510f4 and `RB_THREAD_LOCAL_SPECIFIER` is declared instead.

So on Ruby 3.4 we rely on the available out of the box `RB_THREAD_LOCAL_SPECIFIER` and on earlier versions the previous logic is used. This way thread-local declarations should work on both existing Ruby versions and Ruby 3.4.